l2geth/internal/ethapi/api.go

```diff
@@ -1040,29 +1040,30 @@ func DoEstimateGas(ctx context.Context, b Backend, args CallArgs, blockNrOrHash
 
 	// 2a. fetch the data price, depends on how the sequencer has chosen to update their values based on the
 	// l1 gas prices
-	dataPrice, err := b.SuggestDataPrice(ctx)
+	l1GasPrice, err := b.SuggestDataPrice(ctx)
 	if err != nil {
 		return 0, err
 	}
 
 	// 2b. fetch the execution gas price, by the typical mempool dynamics
-	executionPrice, err := b.SuggestExecutionPrice(ctx)
+	l2GasPrice, err := b.SuggestExecutionPrice(ctx)
 	if err != nil {
 		return 0, err
 	}
 
-	// 3. calculate the fee and normalize by the default gas price
 	var data []byte
 	if args.Data == nil {
 		data = []byte{}
 	} else {
 		data = *args.Data
 	}
-	fee := core.CalculateRollupFee(data, uint64(gasUsed), dataPrice, executionPrice).Uint64() / defaultGasPrice
-	if fee < 21000 {
-		fee = 21000
+	// 3. calculate the fee
+	l2GasLimit := new(big.Int).SetUint64(uint64(gasUsed))
+	fee, err := core.CalculateRollupFee(data, l1GasPrice, l2GasLimit, l2GasPrice)
+	if err != nil {
+		return 0, err
 	}
```

Comment:
We should consider doing a `fee.IsUint64` check here
https://golang.org/pkg/math/big/#Int.IsUint64

Reply:
Do this and return an error message saying fee too high